src/mavsdk/plugins/log_streaming/log_streaming_impl.cpp

```diff
@@ -0,0 +1,326 @@
+                   << ", sequence: " << logging_data.sequence;
+    }
+
+    std::lock_guard<std::mutex> lock(_mutex);
+    check_sequence(logging_data.sequence);
+
+    if (logging_data.first_message_offset == std::numeric_limits<uint8_t>::max()) {
+        _ulog_data.insert(
+            _ulog_data.end(), logging_data.data, logging_data.data + logging_data.length);
+    } else {
+        if (logging_data.first_message_offset > 0) {
+            _ulog_data.insert(
+                _ulog_data.end(),
+                logging_data.data,
+                logging_data.data + logging_data.first_message_offset);
+            process_message();
+            _ulog_data.clear();
+        }
+        _ulog_data.insert(
+            _ulog_data.end(),
+            logging_data.data + logging_data.first_message_offset,
+            logging_data.data + logging_data.length);
+    }
+}
+
+void LogStreamingImpl::process_logging_data_acked(const mavlink_message_t& message)
+{
+    mavlink_logging_data_acked_t logging_data_acked;
+    mavlink_msg_logging_data_acked_decode(&message, &logging_data_acked);
+
+    if (logging_data_acked.target_system != _system_impl->get_own_system_id() ||
+        logging_data_acked.target_component != _system_impl->get_own_component_id()) {
+        LogWarn() << "Logging data acked with wrong target "
+                  << std::to_string(logging_data_acked.target_system) << '/'
+                  << std::to_string(logging_data_acked.target_component) << " instead of "
+                  << std::to_string(_system_impl->get_own_system_id()) << '/'
+                  << std::to_string(_system_impl->get_own_component_id());
+        return;
+    }
+
+    _system_impl->queue_message(
+        [target_system = message.sysid,
+         target_component = message.compid,
+         sequence = logging_data_acked.sequence](MavlinkAddress mavlink_address, uint8_t channel) {
+            mavlink_message_t response_message;
+            mavlink_msg_logging_ack_pack_chan(
+                mavlink_address.system_id,
+                mavlink_address.component_id,
+                channel,
+                &response_message,
+                target_system,
+                target_component,
+                sequence);
+            return response_message;
+        });
+
+    std::lock_guard<std::mutex> lock(_mutex);
+    check_sequence(logging_data_acked.sequence);
+
+    if (_debugging) {
+        LogInfo() << "Received logging data acked with len: "
+                  << std::to_string(logging_data_acked.length)
+                  << ", first message: " << std::to_string(logging_data_acked.first_message_offset)
+                  << ", sequence: " << logging_data_acked.sequence;
+    }
+
+    if (logging_data_acked.first_message_offset == std::numeric_limits<uint8_t>::max()) {
+        _ulog_data.insert(
+            _ulog_data.end(),
+            logging_data_acked.data,
+            logging_data_acked.data + logging_data_acked.length);
```

Comment:
Just to be on the safe side, you could add a check if `logging_data_acked.length` exceeds the `logging_data_acked.data` size.

Reply:
Good catch. I'll add that. So that's basically for sneaky values between 250 and 254.